### PR TITLE
[SPARK-34965][BUILD] Remove .sbtopts that duplicately sets the default memory

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

This PR removes `.sbtopts` (added in https://github.com/apache/spark/pull/29286) that duplicately sets the default memory. The default memories are set:

https://github.com/apache/spark/blob/3b634f66c3e4a942178a1e322ae65ce82779625d/build/sbt-launch-lib.bash#L119-L124

### Why are the changes needed?

This file disables the memory option from the `build/sbt` script:

```bash
./build/sbt -mem 6144
```
```
.../jdk-11.0.3.jdk/Contents/Home as default JAVA_HOME.
Note, this will be overridden by -java-home if it is set.
Error occurred during initialization of VM
Initial heap size set to a larger value than the maximum heap size
```

because it adds these memory options at the last:

```bash
/.../bin/java -Xms6144m -Xmx6144m -XX:ReservedCodeCacheSize=256m -Xmx4G -Xss4m -jar build/sbt-launch-1.5.0.jar
```

and Java respects the rightmost memory configurations.

### Does this PR introduce _any_ user-facing change?

No, dev-only.

### How was this patch tested?

Manually ran SBT. It will be tested in the CIs in this Pr.

### CODE REVIEW

**Review by cloud-fan — APPROVED** [review]
makes sense

**Review by srowen — APPROVED** [review]

File: build/sbt-launch-lib.bash [hunk 1]
```diff
@@ -17,7 +17,7 @@ declare -a java_args
 declare -a scalac_args
 declare -a sbt_commands
 declare -a maven_profiles
-declare sbt_default_mem=2048
+declare sbt_default_mem=4096
```
Comment (HyukjinKwon) [inline build/sbt-launch-lib.bash:20]: So here was the default place to set the memory

**Review by dongjoon-hyun — APPROVED** [review]

**Comment by dongjoon-hyun** [comment]
Thank you, @HyukjinKwon and all.
Merged to master/3.1.

**Comment by HyukjinKwon** [comment]
Thanks!

**Pull request closed without merging**